### PR TITLE
Add rich comparison methods for Text

### PR DESCRIPTION
**Pull request by msempere**
## Type of changes

- [ ] Bug fix
- [x] New feature
- [ ] Documentation / docstrings
- [ ] Tests
- [ ] Other

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [x] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [x] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description
This pull request adds rich comparison methods to the `rich.Text` class in order to address the issue #2921. The added methods enable comparison of rich text objects using Python's rich comparison operators like <, <=, ==, !=, >= and >. These methods make it easy for users to compare rich text objects and perform conditional logic on them. With this addition, users can now leverage the full power of the Python programming language with rich text objects.
Closes https://github.com/Textualize/rich/issues/2921

### CODE REVIEW

File: rich/text.py [hunk 1]
```diff
@@ -171,6 +171,31 @@ def __eq__(self, other: object) -> bool:
...
+        if not isinstance(other, Text):
+            return NotImplemented
+        return self.plain <= other.plain
+
+    def __gt__(self, other: object) -> bool:
+        if not isinstance(other, Text):
+            return NotImplemented
+        return self.plain > other.plain
+
+    def __ne__(self, other: object) -> bool:
```
[the remark this answers is not in the document]
Comment (msempere) [reply rich/text.py:189]: Mmmm I see. I'll address that to make `__ne__` the logical inverse.

**Comment by msempere** [comment]
@willmcgugan is there anything else you would like me to address here?

File: rich/text.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (msempere) [reply rich/text.py:189]: From the docs:

```For __ne__(), by default it delegates to __eq__() and inverts the result unless it is NotImplemented```

I'll remove the `__ne__` following that.

[the remark this answers is not in the document]
Comment (msempere) [reply rich/text.py]: From the [docs](https://docs.python.org/3/reference/datamodel.html#object.__ne__):

```
By default, __ne__() delegates to __eq__() and inverts the result unless it is NotImplemented.
```

So I'm removing the unnecessary `__ne__` implementation.

**Comment by msempere** [comment]
@willmcgugan any chance you could review this PR please? thanks

**Comment by aaronst** [comment]
@msempere was this deleted and closed for any particular reason? It'd be great to get this merged.

**Comment by msempere** [comment]
> @msempere was this deleted and closed for any particular reason? It'd be great to get this merged.

@aaronst Since there was no participation or interaction on this pull request, I've opted to close it.

**Comment by aaronst** [comment]
@msempere I'd encourage you to re-open and give @willmcgugan and folks more time. They're quite busy but they will eventually review the PR, I think this would be a nice addition.

**Comment by msempere** [comment]
> @msempere I'd encourage you to re-open and give @willmcgugan and folks more time. They're quite busy but they will eventually review the PR, I think this would be a nice addition.

@aaronst reopening this PR. Thanks for the ping.

**Pull request closed without merging**